src/SignUpForm.js

```diff
@@ -24,56 +24,48 @@ class signUpForm extends React.Component {
     });
   }
 
-  handleSubmit(event) {
-    event.preventDefault();
+  fetchParams() {
     let data = {
       username: this.state.username,
       email: this.state.email,
       full_name: this.state.full_name,
       password: this.state.password
     }
-    signUp(data)
-      .then(res => this.props.setSignedIn(res))
-      .then(this.props.setSignedUp())
-      .catch(err => alert(`${data.error}`))
+    return ({
+      method: 'POST',
+      mode: 'cors',
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(data)
+    })
+  }
+
+  signUp(event) {
+    event.preventDefault();
+    fetch(`${BASE_URL}/api/v1/users`, this.fetchParams())
+      .then(res => res.json())
+      .then(res => this.props.setUserData(res))
+      .catch(err => alert('Error with sign up'))
```

Comment:
future changes: actually return the error the backend outputs

Reply:
Agreed